File: Code/Fluent.Java/src/adstack/components/QuattroporteDBWAdapter.java
package adstack.components;

import adstack.models.IComponent;

public class QuattroporteDBWAdapter implements IComponent {

	/* Singleton pattern */
	private static QuattroporteDBWAdapter _Instance = null;

	public static QuattroporteDBWAdapter GetInstance() {
		if(_Instance == null) _Instance = new QuattroporteDBWAdapter();
		return _Instance;
	}
	
	private QuattroporteDBWAdapter() {
		
	}
	
	/* As IComponent */
	
	@Override
	public void start() {
        System.out.println("[QuattroporteDBWAdapter] Starting.");
	}

	@Override
	public void stop() {
        System.out.println("[QuattroporteDBWAdapter] Stopping.");
	}

}

File: Code/dotNetBasics/ConsoleApp/ADStack/Components/QuattroporteDBWAdapter.cs
using ConsoleApp.ADStack.Models;

namespace ConsoleApp.ADStack.Components
{
    /// <summary>
    /// Maserati 4porte Drive-by-Wire system.
    /// Do you remember the Adapter and Singleton Patterns?
    /// </summary>
    internal class QuattroporteDBWAdapter : IComponent
    {
        private static QuattroporteDBWAdapter _Instance = null;
        private QuattroporteDBWAdapter()
        {
        }

        public static QuattroporteDBWAdapter GetInstance()
        {
            if (_Instance == null) _Instance = new QuattroporteDBWAdapter();
            return _Instance;
        }

        public void Start()
        {
            Console.WriteLine("[QuattroporteDBWAdapter] Starting.");
        }

        public void Stop()
        {
            Console.WriteLine("[QuattroporteDBWAdapter] Stopping.");
        }
    }
 }
